### DISCUSSION TITLE
How to add CDN in laravel using Vite?

### DISCUSSION THREAD

**Discussion by DevMushref** [filamentphp/filament#5493] [Help] [answered]
I am trying to include two scripts, one is a cdn:
`<script src="https://cdn.onesignal.com/sdks/OneSignalSDK.js" async=""></script>`

The other is OneSignal code:

```
<script>
        window.OneSignal = window.OneSignal || [];
        OneSignal.push(function() {
            OneSignal.init({
                appId: "xxxxxxxxxxxxxxxxxxxxxx",
            });
        });
    </script>
```
I added these two inside `app.blade.php` but they don't work (I checked Inspect, the codes aren't rendered/showing). I read more about Vite in the Laravel documentation but couldn't understand the process.

**Comment by ArtMin96** [comment, marked as answer]
If I understood correctly, you need this

```php
use Filament\Facades\Filament;
 
Filament::registerScripts([app(Vite::class)('https://cdn.onesignal.com/sdks/OneSignalSDK.js')]);
```

vite.config.js

```javascript
export default defineConfig({
    plugins: [
        laravel({
            input: [
                'https://cdn.onesignal.com/sdks/OneSignalSDK.js' // I don't know if it is allowed to add cdn here or not
                //...
            ],
            refresh: true,
        }),
```

Or check the docs: [https://filamentphp.com/docs/2.x/admin/appearance#including-frontend-assets](https://filamentphp.com/docs/2.x/admin/appearance#including-frontend-assets)

**Reply by DevMushref** [reply]
Yes, this is what I wanted, Thank you for your time.
